client/src/http_client.rs

```diff
@@ -27,6 +27,43 @@ impl ImmuxDBHttpClient {
 }
 
 impl ImmuxDBClient<HttpClientResult> for ImmuxDBHttpClient {
+    fn get_all_groupings(&self) -> HttpClientResult {
+        let url = format!("http://{}/{}", &self.host, Constants::URL_GROUPING_KEY_WORD,);
+
+        let mut response = self.client.get(&url).send()?;
+        let status_code = response.status();
+
+        match response.text() {
+            Ok(text) => Ok((status_code, text)),
+            Err(error) => Err(ImmuxDBHttpClientError::Reqwest(error.into())),
+        }
+    }
+
+    fn remove_groupings(&self, groupings: &[GroupingLabel]) -> HttpClientResult {
+        if groupings.is_empty() {
+            return Err(ImmuxDBHttpClientError::Everything);
+        }
+
+        let url = format!("http://{}/{}", &self.host, Constants::URL_GROUPING_KEY_WORD,);
+        let groupings =
+            groupings[1..]
+                .iter()
+                .fold(format!("{}", groupings[0]), |mut acc, grouping| {
+                    acc.push_str("\r\n");
+                    let grouping_str = format!("{}", grouping);
+                    acc.push_str(&grouping_str);
+                    return acc;
+                });
```

Comment:
Do we have to use explicit folding and concatenation?

Can we go for something like
```
        let grouping_names = groupings
            .iter()
            .map(|g| g.to_string())
            .collect::<Vec<String>>()
            .join("\r\n");

```

Reply:
good point, done